frontend/src/app/libs/common-components/components/input/input.component.ts

```diff
@@ -0,0 +1,283 @@
+		this.errorsArray = [];
+		this.invokeSyncValidators();
+		this.invokeAsyncValidators().
+			then(() => {
+				if (Boolean(this.errorsArray?.length)) {
+					this.formControl.setErrors({error: 'formControl has error'});
+					this.changeDetector.detectChanges();
+				} else {
+					this.formControl.setErrors(null);
+					this.changeDetector.detectChanges();
+				}
+			});
+	}
+
+	private setSyncValidator(): void {
+		this.syncValidators = [].concat(this.inputValidators, this.changeValidators)
+		.filter((validator: ValidatorFn | null) => validator);
+	}
+	private setAsyncValidator(): void {
+		this.asyncValidators = [].concat(this.inputValidatorsAsync, this.changeValidatorsAsync)
+		.filter((validator: ValidatorFn | null) => validator);
+	}
+
+	private invokeSyncValidators(): void {
+		if (Boolean(this.syncValidators?.length)) {
+			this.syncValidators.map((validator: ValidatorFn ) => {
+				const error: ValidationErrors = validator(this.formControl);
+				// this.formControl.setErrors(error);
+				if (error?.message) {
+				this.getCustomErrorsArray(error);
+				}
+
+			});
+		}
+	}
+
+	private async invokeAsyncValidators(): Promise<any> {
+		return new Promise((resolve: (value?: any) => void) => {
+			if (Boolean (this.asyncValidators?.length)) {
+				this.asyncValidators.map(async (validator: AsyncValidatorFn) => {
+					let asyncError: Promise<ValidationErrors | null> | Observable<ValidationErrors | null> = validator(this.formControl);
+					asyncError = this.promiseToObservable(asyncError);
+					asyncError.pipe(take(1)).subscribe((error: ValidationErrors) => {
+						if (error?.message && !this.errorsArray?.includes(error.message)) {
+							this.getCustomErrorsArray(error);
+						}
+						resolve();
+					});
+				});
+			} else {
+				resolve();
+			}
+		});
+
+	}
+
+	private isAsyncErrorPromise(error: Promise<ValidationErrors | null> | Observable<ValidationErrors | null>): boolean {
+		return (error as Promise<ValidationErrors>).then !== undefined;
+	}
+
+	private promiseToObservable(error: Promise<ValidationErrors | null> | Observable<ValidationErrors | null>): Observable<ValidationErrors> {
+
+		if (this.isAsyncErrorPromise(error)) {
+			return fromPromise(error as Promise<ValidationErrors>);
+		}
+		return error as Observable<ValidationErrors>;
+	}
+
+	private checkInlineValidators(): void {
+		if (this.formControl.validator) {
+			this.formControl.clearValidators();
+			throw new Error(this.errorOnInlineValidators);
+		}
+		if (this.formControl.asyncValidator) {
+			this.formControl.clearAsyncValidators();
+			throw new Error(this.errorOnInlineValidators);
+		}
+	}
+
+	private subscribeToChanges(): void {
+		this.formControl.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(() => {
+			if (Boolean(this.errorsArray?.length)) {
+				this.formControl.setErrors({error: 'formControl has error'});
+			}
+		});
+	}
+
+	private getCustomErrorsArray(error: ValidationErrors): void {
+		if (typeof(error.message === 'string')) {
```

Comment:
no need to check for string, just make sure it's not null/undefined

Reply:
its just to avoid unnecessary types of errors as boolean or even object